[feat] User Profile Image Upload with Optimization and Resizing using ImageKit.io

IiamV opened the pull request:
This PR is linked with issues #7 and #4

updateUserProfile
### Form-data Input
```json
{
   "full_name": "text",
   "birth_date": "text",
   "image_file": "file"
}
```
- `updateUserProfile`:The function validates file types (JPG, JPEG, PNG, WebP), generates unique filenames (`profile_[USER_ID].[EXTENSION]`), and uses [ImageKit.io](https://imagekit.io/) for [resizing](https://imagekit.io/docs/image-resize-and-crop#width---w) (mandatory) and optional [compression](https://imagekit.io/docs/image-optimization#quality---q) (triggered if the file exceeds the `optimizationSizeLimit`, reverting to the original if optimized size increases), the [free plan](https://imagekit.io/plans/) offering 25GB free monthly bandwidth. The workflow uploads the image to ImageKit, processes it, deletes the temporary ImageKit file to save space, removes the old user profile image from `profile_picture_url` in `user_details`, and updates the database with the new image URL, `full_name`, and/or `birth_date` as provided, returning a success response immediately while deferring processing to the background.

IiamV committed:
updateTicketStatus

update ticket status function, not cleaned, correct the link in .env

IiamV committed:
db functions

updateTicketStatus + createAppointment

IiamV committed:
order arrangements

move createAppointment and updateTicketStatus out of supabase create_ticket edge func

IiamV committed:
undescribe

-comments

IiamV committed:
fixedLinking

fixed the bug with create_ticket and createAppointment

IiamV committed:
clean and optimized

IiamV committed:
remove stuff

unnecessary stuff related to deploying it onto the web

IiamV committed:
add comments

IiamV committed:
test comment

IiamV committed:
comment order

IiamV committed:
Add updateUserProfile function with image processing

Introduces a new Supabase Edge Function 'updateUserProfile' for handling user profile updates, including image upload, validation, resizing, and optimization using ImageKit. Updates the Supabase config to register the new function, adds environment variable templates, and includes configuration files for Deno and npm.

IiamV committed:
Improve code comments

IiamV committed:
Delete .env

IiamV committed:
Merge pull request #1 from IiamV/master

[feat] Added updateUserProfile function

IiamV commented:
> @maik205 

Does this solution address the mentioned issues? If yes, I'll be making another PR that adds the necessary policies for interacting with the `profile-images` bucket.

maik205 commented on supabase/functions/updateUserProfile/index.ts:
```diff
@@ -0,0 +1,438 @@
+      headers: { ...corsHeaders, 'Content-Type': 'application/json' } 
+    }
+  )
+}
```
It's possible to write these inline since this code won't be reused often.

IiamV replied:
> @maik205 

Does the latest commit look good to you?

maik205 reviewed (changes requested):
Can you write used-once-only functions into the serve block, thank you!

IiamV committed:
Refactor response handling and profile update logic

Removed standardized responses and replaced them with inline Response objects for error and success cases. Updated profile update logic, now using a database function for user details.

IiamV commented:
> @maik205 

Also I've edit the update user logic, it's now using database function instead of manually doing it.

maik205 commented:
Awesome, LGTM
